modules/navier_stokes/src/fvkernels/PINSFVMomentumDiffusion.C

```diff
@@ -45,12 +45,17 @@ PINSFVMomentumDiffusion::computeStrongResidual()
 #ifdef MOOSE_GLOBAL_AD_INDEXING
   using namespace Moose::FV;
 
+  const bool has_elem = (_face_type == FaceInfo::VarFaceNeighbors::ELEM ||
+                         _face_type == FaceInfo::VarFaceNeighbors::BOTH);
+  const bool has_neighbor = (_face_type == FaceInfo::VarFaceNeighbors::NEIGHBOR ||
+                             _face_type == FaceInfo::VarFaceNeighbors::BOTH);
+
   const auto elem_face = elemFromFace();
   const auto neighbor_face = neighborFromFace();
   const auto mu_elem = _mu(elem_face);
   const auto mu_neighbor = _mu(neighbor_face);
-  const auto eps_elem = _eps(elem_face);
-  const auto eps_neighbor = _eps(neighbor_face);
+  const auto eps_elem = has_elem ? _eps(elem_face) : _eps(neighbor_face);
```

Comment:
Isn't the single-sided evaluation designed for this? Or the two regions are not separated by a sideset? I suppose we could check in the PWBBLF in the evaluate(FaceArg) if it has lambdas for both the elem.block_id() and neighbor.block_id() if not, it could fall back to an extrapolatedBoundary evaluation.

Reply:
it is at boundaries. but not inside the domain.
I dont want SSF to be interpolations inside the domain either. So we do want a FA here.

FA on the neighbor side just didnt make too much sense either, it was out of the block restriction seemingly.

Reply:
When you are on the boundary of a block-restricted physics, I do think that SSF makes sense conceptually